sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/RuleExecutor.scala

```diff
@@ -155,8 +162,9 @@ abstract class RuleExecutor[TreeType <: TreeNode[_]] extends Logging {
         if (iteration > batch.strategy.maxIterations) {
           // Only log if this is a rule that is supposed to run more than once.
           if (iteration != 2) {
-            val message = s"Max iterations (${iteration - 1}) reached for batch ${batch.name}"
-            if (Utils.isTesting) {
+            val message = s"Max iterations (${iteration - 1}) reached for batch ${batch.name}, " +
+              s"increasing the value of '${SQLConf.ANALYZER_MAX_ITERATIONS.key}'."
```

Comment:
Actually, this change is not right. The optimizer will also issue this message.

Reply:
Opps, sorry.
I updated the logic in cdc48795b8ad39359186830a7cc0c5be3dc169aa. Extend abstract class `Strategy` to support setting different hint setting key in Analyzer or Optimizer.
Would you please take time to review? Thanks!